velox/exec/tests/VeloxIn10MinDemo.cpp

```diff
@@ -0,0 +1,248 @@
+  core::TypedExprPtr parseExpression(
+      const std::string& text,
+      const RowTypePtr& rowType) {
+    auto untyped = parse::parseExpr(text);
+    return core::Expressions::inferTypes(untyped, rowType, execCtx_->pool());
+  }
+
+  /// Compile typed expression tree into an executable ExprSet.
+  std::unique_ptr<exec::ExprSet> compileExpression(
+      const std::string& expr,
+      const RowTypePtr& rowType) {
+    std::vector<core::TypedExprPtr> expressions = {
+        parseExpression(expr, rowType)};
+    return std::make_unique<exec::ExprSet>(
+        std::move(expressions), execCtx_.get());
+  }
+
+  /// Evaluate an expression on one batch of data.
+  VectorPtr evaluate(exec::ExprSet& exprSet, const RowVectorPtr& input) {
+    exec::EvalCtx context(execCtx_.get(), &exprSet, input.get());
+
+    SelectivityVector rows(input->size());
+    std::vector<VectorPtr> result(1);
+    exprSet.eval(rows, &context, &result);
+    return result[0];
+  }
+
+  /// Make TPC-H split to add to TableScan node.
+  exec::Split makeTpchSplit() const {
+    return exec::Split(std::make_shared<connector::tpch::TpchConnectorSplit>(
+        kTpchConnectorId));
+  }
+
+  /// Run the demo.
+  void run();
+
+  std::shared_ptr<core::QueryCtx> queryCtx_{core::QueryCtx::createForTest()};
+  std::unique_ptr<core::ExecCtx> execCtx_{
+      std::make_unique<core::ExecCtx>(pool_.get(), queryCtx_.get())};
+};
+
+void VeloxIn10MinDemo::run() {
+  // Let’s create two vectors of 64-bit integers.
+  auto a = makeFlatVector<int64_t>({0, 1, 2, 3, 4, 5});
+  auto b = makeFlatVector<int64_t>({0, 5, 10, 15, 20, 25});
+
+  auto data = makeRowVector({"a", "b"}, {a, b});
+
+  std::cout << std::endl
+            << "> vectors a and b: " << data->toString() << std::endl;
+  std::cout << data->toString(0, data->size()) << std::endl;
+
+  // Now, let’s compute a sum of 'a' and 'b' by evaluating 'a + b' expression.
+
+  // First we need to parse the expression into a fully typed expression tree.
+  // Then, we need to compile it into an executable ExprSet.
+  auto exprSet = compileExpression("a + b", asRowType(data->type()));
+
+  // Let's print out the ExprSet:
+  std::cout << std::endl << "> 'a + b' expression:" << std::endl;
+  std::cout << exprSet->toString() << std::endl;
+
+  // Now we are ready to evaluate the expression on a batch of data.
+  auto c = evaluate(*exprSet, data);
+
+  auto abc = makeRowVector({"a", "b", "c"}, {a, b, c});
+
+  std::cout << std::endl << "> a, b, a + b: " << abc->toString() << std::endl;
+  std::cout << abc->toString(0, c->size()) << std::endl;
+
+  // Let's try a slightly more complex expression: `3 * a + sqrt(b)`.
+  exprSet = compileExpression("2 * a + b % 3", asRowType(data->type()));
+
+  std::cout << std::endl << "> '2 * a + b % 3' expression:" << std::endl;
+  std::cout << exprSet->toString() << std::endl;
+
+  auto d = evaluate(*exprSet, data);
+
+  auto abcd = makeRowVector({"a", "b", "c", "d"}, {a, b, c, d});
```

Comment:
I'm a bit lost about the use of vector "abcd" here. Why is it not sufficient to print from the vector d itself ?

Reply:
I feel it is nicer to print out the inputs alongside the outputs of the computation. It is also possible to print just 'd'.